src/accumulator.jl

```diff
@@ -9,30 +9,22 @@ end
 Accumulator(::Type{T}, ::Type{V}) where {T,V<:Number} = Accumulator{T,V}(Dict{T,V}())
```

Comment:
As with other constructors, we should deprecate this constructor, and add a constructor for `Accumulator{T,V<:Integer}()`.

Probably not that urgent, as the primary interface to this seems to be the `counter` function.  Can be done in a separate PR (unless you want to do it here).